### ISSUE TITLE
in arrow branch - tidyCovariates fails if it removes rare, but not redundant covariates

### ISSUE THREAD

**Issue by egillax** [OHDSI/FeatureExtraction#175]
I'm testing arrow in PLP and came upon an edgecase which breaks tidyCovariates.

To reproduce:

```R
createCovariate <- function(i, analysisId) {
    return(tibble(covariateId = rep(i * 1000 + analysisId, i),
                          covariateValue = rep(1,i)))
  }
  covariates <- lapply(1:10, createCovariate, analysisId = 1)
  covariates <- do.call("rbind", covariates)
  covariates$rowId <- 1:nrow(covariates)
  metaData <- list(populationSize = nrow(covariates))
  frequentCovariate <- createCovariate(40, analysisId = 2)
  frequentCovariate$rowId <- sample.int(metaData$populationSize, nrow(frequentCovariate), replace = FALSE)
  infrequentCovariate <- createCovariate(1, analysisId = 3)
  infrequentCovariate$rowId <- sample.int(metaData$populationSize, nrow(infrequentCovariate), replace = FALSE)
  covariates <- rbind(covariates, frequentCovariate, infrequentCovariate)

  covariateRef <- tibble(covariateId = c(1:10 * 1000 + 1, 40002, 1003),
                                 analysisId = c(rep(1, 10), 2, 3))

  covariateData <- Andromeda::andromeda(covariates = covariates,
                                        covariateRef = covariateRef)
  attr(covariateData, "metaData") <- metaData
  class(covariateData) <-  c("CovariateData", "Andromeda")

  tidy <- tidyCovariateData(covariateData, minFraction = 0.1, normalize = FALSE, removeRedundancy = FALSE)
  ```
This is because if it doesn't remove rare covariates the ```ignoreCovariateIds``` variable is ```NULL``` when it comes to the filtering expression [here](https://github.com/OHDSI/FeatureExtraction/blob/arrow/R/Normalization.R#L147). This seems to break arrow. I did implement a workaround locally by using the line 
```R
if (is.null(ignoreCovariateIds)) ignoreCovariateIds <- c(-1)
```
before this filtering and it seems to work.

**Comment by schuemie** [comment]
It would be good to better understand what is causing this behavior. Unfortunately, in R, there's no difference between an empty vector and NULL. Does this mean arrow can't do an `%in%` filter with an empty set? That would be a potential source of a lot of bugs that can only be detected at runtime.

Base R has no issues with empty sets:
```r
1:3 %in% c()
# [1] FALSE FALSE FALSE
```

**Comment by egillax** [comment]
You're right. I did look a bit into this. Seems arrow is unable to detect correctly the type of ```c()```.  However the underlying compute function arrow uses for this operation does accepts nulls, so if I use some lower level functions to explicitly tell arrow ```c()``` is null it works. I think this is a bug since base R handles this but the arrow replacement which is supposed to be "drop in" doesn't.

I created an issue about it in the arrow issue tracker [here](https://issues.apache.org/jira/browse/ARROW-17543)
